lib/Runtime/Base/FunctionBody.cpp

```diff
@@ -570,6 +577,291 @@ namespace Js
+        this->m_hasBeenParsed = true;
+
+        Assert(!proxy->GetUtf8SourceInfo() || m_uScriptId == proxy->GetUtf8SourceInfo()->GetSrcInfo()->sourceContextInfo->sourceContextId);
+
+        // Sync entryPoints changes to etw rundown lock
+        CriticalSection* syncObj = scriptContext->GetThreadContext()->GetEtwRundownCriticalSection();
+        this->entryPoints = RecyclerNew(scriptContext->GetRecycler(), FunctionEntryPointList, scriptContext->GetRecycler(), syncObj);
+
+        this->AddEntryPointToEntryPointList(this->GetDefaultFunctionEntryPointInfo());
+
+        Assert(this->GetDefaultEntryPointInfo()->jsMethod != nullptr);
+
+        InitDisableInlineApply();
+        InitDisableInlineSpread();
+    }
+
+    bool FunctionBody::InterpretedSinceCallCountCollection() const
+    {
+        return this->interpretedCount != this->lastInterpretedCount;
+    }
+
+    void FunctionBody::CollectInterpretedCounts()
+    {
+        this->lastInterpretedCount = this->interpretedCount;
+    }
+
+    void FunctionBody::IncrInactiveCount(uint increment)
+    {
+        this->inactiveCount = UInt32Math::Add(this->inactiveCount, increment);
+    }
+
+    bool FunctionBody::IsActiveFunction(ActiveFunctionSet * pActiveFuncs) const
+    {
+        return !!pActiveFuncs->Test(this->GetFunctionNumber());
+    }
+
+    bool FunctionBody::TestAndUpdateActiveFunctions(ActiveFunctionSet * pActiveFuncs) const
+    {
+        return !!pActiveFuncs->TestAndSet(this->GetFunctionNumber());
+    }
+
+    void FunctionBody::UpdateActiveFunctionSet(ActiveFunctionSet *pActiveFuncs) const
+    {
+        if (this->TestAndUpdateActiveFunctions(pActiveFuncs))
+        {
+            return;
+        }
+        FunctionCodeGenRuntimeData **data = this->GetCodeGenRuntimeData();
+        if (data == nullptr)
+        {
+            return;
+        }
+        for (uint i = 0; i < this->GetProfiledCallSiteCount(); i++)
+        {
+            for (FunctionCodeGenRuntimeData *inlineeData = data[i]; inlineeData; inlineeData = inlineeData->GetNext())
+            {
+                inlineeData->GetFunctionBody()->UpdateActiveFunctionSet(pActiveFuncs);
+            }
+        }
+    }
+
+    bool FunctionBody::DoRedeferFunction(uint inactiveThreshold) const
+    {
+        bool isJitCandidate = false;
+
+        if (!(this->GetFunctionInfo()->GetFunctionProxy() == this &&
+              this->CanBeDeferred() &&
+              this->GetByteCode() &&
+              this->GetCanDefer()))
+        {
+            return false;
+        }
+
+        if (!PHASE_FORCE(Js::RedeferralPhase, this) && !PHASE_STRESS(Js::RedeferralPhase, this))
+        {
+            uint inactiveCount;
+            auto fn = [&](){ inactiveCount = 0xFFFFFFFF; };
+            inactiveCount = UInt32Math::Mul(this->GetInactiveCount(), this->GetCompileCount(), fn);
+            if (inactiveCount < inactiveThreshold)
```

Comment:
>inactiveCount < inactiveThreshold [](start = 16, length = 33)

The idea here is that if a function was parsed after having been redeferred, make it less likely to be deferred again, right? If so, shouldn't we multiply the threshold by the compileCount?